### PR TITLE
[Blog] Add 'Converting Docker Compose files and Kubernetes Manifests to Meshery Pattern Files'

### PR DESCRIPTION
**Pull request by gr455**
**Description**
This PR adds the 'Converting Docker Compose files and Kubernetes Manifests to Meshery Pattern Files' blog post

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+services:
+  istio:
+    type: IstioMesh
+    namespace: istio-system
+    settings:
+      version: 1.14
+      profile: demo
+```
+</p>
+<p>(Meshery pattern files are still under active development. Refer [the Service Mesh Patterns Design Spec](https://docs.google.com/document/d/1B2N78EdRiZF-yVo1-HY3syppwBBDumgMuYg6seD-AJ4) for more information)</p>
```
Comment (debo19) [inline src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:44]: You'll have to use the `<a>` tag here

File: src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import dashboardImg from "./dashboard.png";
+import applicationsPageImg from "./applications_page.png";
+import importAppImg from "./import_app.png";
+import clickDeployImg from "./click_deploy.png";
+import requestPayloadImg from "./request_payload.png";
+
+<BlogWrapper>
+
+<h3>Meshery Pattern Files</h3>
+<p>Meshery uses service mesh patterns that act as configurations of service meshes or applications to be deployed through Meshery. These configurations are written in a YAML format in a pattern file. Here's a small example pattern file to deploy a demo deployment of Istio through Meshery:
```
Comment (debo19) [inline src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:30]: Link Meshery to `/service-mesh-management/meshery`
Service mesh Patterns to `/learn/service-mesh-books/service-mesh-patterns`

File: src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    namespace: istio-system
+    settings:
+      version: 1.14
+      profile: demo
+```
+</p>
+<p>(Meshery pattern files are still under active development. Refer [the Service Mesh Patterns Design Spec](https://docs.google.com/document/d/1B2N78EdRiZF-yVo1-HY3syppwBBDumgMuYg6seD-AJ4) for more information)</p>
+
+<h3>Convert Docker Compose / Kubernetes Manifests to Pattern Files</h3>
+<p>Although as a user, you can deploy applications specified as Kubernetes Manifests or Docker Compose files directly through Meshery, you might want to convert them to Meshery Pattern Files. This tutorial can help you do just that
```
Comment (debo19) [inline src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:47]:
```suggestion
<p>Although as a user, you can deploy applications specified as Kubernetes Manifests or Docker Compose files directly through Meshery, you might want to convert them to Meshery Pattern Files. This tutorial can help you do just that.
```

File: src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    <li>
+    Open the Meshery UI. On the home page you should see the dashboard
+    <img src = {dashboardImg} className="image-center-shadow" alt="Screenshot of Meshery Dashboard" />
+    </li><br />
+    <li>
+    Navigate to `Configuration > Applications` from the navbar on the side
+    <img src = {applicationsPageImg} className="image-center-shadow" alt="Screenshot of Meshery Applications page" />
+    </li><br />
+    <li>
+    Click on `Import Application` and choose upload your Docker Compose file or Kubernetes Manifest
```
Comment (debo19) [inline src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:58]: If you want these to be code blocks, use `<code>Import Application</code>`

**Review by debo19 — CHANGES_REQUESTED** [review]
This is great! @gr455 🎉

File: src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx (hunk 1 above)
Comment (gr455) [reply src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:44]: Whoops, copied this directly from the markdown

**Comment by l5io** [comment]
🚀 Preview for commit 69c3f2f8c3edfeb8779d72d14c5e6cf7d32a419e at: https://62b9fffb19a0c60a98c7561c--layer5.netlify.app

File: src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import dashboardImg from "./dashboard.png";
+import applicationsPageImg from "./applications_page.png";
+import importAppImg from "./import_app.png";
+import clickDeployImg from "./click_deploy.png";
+import requestPayloadImg from "./request_payload.png";
+
+<BlogWrapper>
+
+<h3>Meshery Pattern Files</h3>
+<p><a href="/service-mesh-management/meshery">Meshery</a> uses <a href="/learn/service-mesh-books/service-mesh-patterns">Service Mesh Patterns</a> that act as configurations of service meshes or applications to be deployed through Meshery. These configurations are written in a YAML format in a pattern file. Here's a small example pattern file to deploy a demo deployment of Istio through Meshery:
```
Comment (debo19) [inline src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:30]:
```suggestion
<p><Link href="/service-mesh-management/meshery">Meshery</a> uses <Link href="/learn/service-mesh-books/service-mesh-patterns">Service Mesh Patterns</a> that act as configurations of service meshes or applications to be deployed through Meshery. These configurations are written in a YAML format in a pattern file. Here's a small example pattern file to deploy a demo deployment of Istio through Meshery:
```

Comment (debo19) [reply src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:30]: Let's import and use the gatsby `Link` component for internal linking. Rest looks good.

File: src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+services:
+  istio:
+    type: IstioMesh
+    namespace: istio-system
+    settings:
+      version: 1.14
+      profile: demo
+```
+</p>
+<p>(Meshery pattern files are still under active development. Refer <a href="https://docs.google.com/document/d/1B2N78EdRiZF-yVo1-HY3syppwBBDumgMuYg6seD-AJ4">the Service Mesh Patterns Design Spec</a> for more information)</p>
```
Comment (debo19) [inline src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:44]: One problem with including the design spec link is that it won't be available to all readers if they don't have access to our drive. Not sure if we should include it. @leecalcote what do you think?

Comment (gr455) [reply src/collections/blog/2022/2022-06-27-converting-docker-compose-files-and-kubernetes-manifests-to-meshery-pattern-files/index.mdx:44]: Okay, I'll remove it then :+1:

**Comment by l5io** [comment]
🚀 Preview for commit d0d9636fc55bb97cddf9f937b8c160fa153b306b at: https://62ba88df832b5421b3609592--layer5.netlify.app

**Comment by leecalcote** [comment]
It would be really neat to be able to show the Istio + SPIRE configuration (Meshery App). // @mchurichi @Revolyssup

**Review by debo19 — APPROVED** [review]
LGTM 🎉 . I think this needs a better thumbnail. We'll assist with that.

**Comment by gr455** [comment]
> LGTM tada . I think this needs a better thumbnail. We'll assist with that.

Thanks a lot! That's the best thumbnail I could come up with :laughing:

**Comment by debo19** [comment]
@leecalcote good to go?

**Comment by leecalcote** [comment]
Will you update the blog to follow this flow:

- introduction to the need for the feature
- Docker Compose section (use logo)
  - example with Docker Compose app.
- Kubernetes section (use logo)
  - example with Kubernetes Manifest.
- Call to Action
- Conclusion

**Comment by leecalcote** [comment]
> > LGTM tada . I think this needs a better thumbnail. We'll assist with that.
> 
> Thanks a lot! That's the best thumbnail I could come up with 😆

Looks real good.

**Comment by gr455** [comment]
> Will you update the blog to follow this flow:
> 
> * introduction to the need for the feature
> * Docker Compose section (use logo)
>   
>   * example with Docker Compose app.
> * Kubernetes section (use logo)
>   
>   * example with Kubernetes Manifest.
> * Call to Action
> * Conclusion

@leecalcote I am not sure what the call for action would be in this. Also the steps for compose file and manifest are the same, do we still want to seperate them into their own sections?

**Comment by debo19** [comment]
> @leecalcote I am not sure what the call for action would be in this. Also the steps for compose file and manifest are the same, do we still want to seperate them into their own sections?

✅ We have call to action added now.
@leecalcote do we still want them in separate sections?

**Comment by l5io** [comment]
🚀 Preview for commit 002d8809a44e81f97608d92cb3ab564e2e4010b8 at: https://630ef4826c785550ae35091a--layer5.netlify.app

**Pull request closed without merging**